### PR TITLE
Fix build warning

### PR DESCRIPTION
**Pull request by AstroAir**
```
static void callErrorHandlingCallback(zip* zipHandle, const std::string& msg, ErrorHandlerCallback* callback) {
        int error_code_zip, error_code_system;
        zip_error_get(zipHandle, &error_code_zip, &error_code_system);
        callErrorHandlingCallbackFunc(msg, error_code_zip, error_code_system, callback);
    }
```
to
```
static void callErrorHandlingCallback(zip* zipHandle, const std::string& msg, ErrorHandlerCallback* callback) {
        auto error_code = zip_get_error(zipHandle);
        callErrorHandlingCallbackFunc(error_code->str, error_code->zip_err, error_code->sys_err, callback);
    }
```
Tested on Msys2(Mingw64)

### CODE REVIEW

File: src/libzippp.cpp [hunk 1]
```diff
@@ -109,9 +109,8 @@ namespace Helper {
     }
 
     static void callErrorHandlingCallback(zip* zipHandle, const std::string& msg, ErrorHandlerCallback* callback) {
-        int error_code_zip, error_code_system;
-        zip_error_get(zipHandle, &error_code_zip, &error_code_system);
-        callErrorHandlingCallbackFunc(msg, error_code_zip, error_code_system, callback);
+        auto error_code = zip_get_error(zipHandle);
```
Comment (ctabin) [inline src/libzippp.cpp:112]:
```suggestion
        zip_error_t* error_code = zip_get_error(zipHandle);
```

**Review by ctabin — CHANGES_REQUESTED** [review]
Hi,
Thanks for the contribution.
Can you please give me the warning ?
Also I'm not sure about the `auto` regarding "old" compilers :thinking:

File: src/libzippp.cpp (hunk 1 above)
Comment (AstroAir) [reply src/libzippp.cpp:112]: Warning:
```
[1/3] Building CXX object CMakeFiles/libzippp.dir/src/libzippp.cpp.obj
E:/msys64/home/test/libzippp-master/src/libzippp.cpp: In function 'void Helper::callErrorHandlingCallback(zip*, const std::string&, void (*)(const std::string&, const std::string&, int, int))':
E:/msys64/home/test/libzippp-master/src/libzippp.cpp:113:22: warning: 'void zip_error_get(zip_t*, int*, int*)' is deprecated: use 'zip_get_error' instead [-Wdeprecated-declarations]
  113 |         zip_error_get(zipHandle, &error_code_zip, &error_code_system);
      |         ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from E:/msys64/home/Qrm/test/libzippp-master/src/libzippp.cpp:40:
E:/msys64/mingw64/include/zip.h:384:63: note: declared here
  384 | ZIP_DEPRECATED("use 'zip_get_error' instead") ZIP_EXTERN void zip_error_get(zip_t *_Nonnull, int *_Nullable, int *_Nullable);
      |                                                               ^~~~~~~~~~~~~
```

Comment (ctabin) [reply src/libzippp.cpp:112]: @AstroAir Thanks for the info. Can you please just change as requested ?

**Comment by AstroAir** [comment]
@ctabin ok

**Comment by ctabin** [comment]
@AstroAir I fixed the pipeline, can you please rebase your branch ? Thanks :pray:

**Comment by AstroAir** [comment]
@ctabin I hava rebased the repo

**Comment by ctabin** [comment]
Awesome, thanks !

**Pull request merged**